airflow/cli/cli_parser.py

```diff
@@ -65,6 +65,9 @@
 
 ALL_COMMANDS_DICT: dict[str, CLICommand] = {sp.name: sp for sp in airflow_commands}
 
+# Hides the subcommands (list) for the given command (key)
+HIDE_SUBCOMMAND = {"db": ["init", "upgrade"]}
```

Comment:
Can we build support for this in ActionCommand instead, maybe a "hide" or "deprecated"? Keeping it where the command is defined seems like it'd be more clear and less likely to be overlooked later.

Reply:
Yes, you are right. Let me make the change.

Reply:
Removed the `HIDE_SUBCOMMAND` and added `hide` keyword in the `ActionCommand`. It's much simpler now. Thanks!